### PR TITLE
Add Realsense Driver handling to HardwareStation

### PR DESCRIPTION
**Pull request by evelyn-fu**
Allow users to specify realsense lcm driver in scenario for HardwareStation in order to use realsense cameras in hardware.

- - -
This change is [Reviewable](https://reviewable.io/reviews/RussTedrake/manipulation/315)

### CODE REVIEW

**Review by RussTedrake — APPROVED** [review]
Thank you!

We already have CameraConfig in the scenarios., which has a name and an lcm_bus (and we use it to configure cameras in these notes... see manipulation/models/clutter.scenarios.yaml for an example).
https://drake.mit.edu/doxygen_cxx/structdrake_1_1systems_1_1sensors_1_1_camera_config.html

Moreover, I don't think anything that you add to the diagram is actually specific to realsense?  So perhaps we already have enough info to just add the lcm receiver, etc, based on the CameraConfig?  

I do think there could be a tension between the the `sensor_id` and the camera name (which we tend to make more readable).  But in that case, maybe we could just bolt on the extra info we need there? e.g. via something like

```
@dc.dataclass
class CameraConfigWithId(CameraConfig):
    """A camera configuration that includes the sensor id (for communicating with e.g. the output of a RealSense driver).

    Args:
        sensor_id: The camera id for lcm.
    """
    sensor_id: str = ""
```

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RussTedrake/manipulation/315#-:-Nyw4bf_4kQlJZ4mkUQT:b-xs40ft)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @evelyn-fu)

**Review by RussTedrake — APPROVED** [review]
btw -- we should also have at least one python test (ideally very minimal) or notebook that runs the code.  just to make sure we know that it works and if/when it ever breaks.

> *[Reviewable](https://reviewable.io/reviews/RussTedrake/manipulation/315#-:-Nywak0P99_sfIkfECt0:b8wqdj8)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @evelyn-fu)

**Comment by evelyn-fu** [comment]
> Thank you!
> 
> We already have CameraConfig in the scenarios., which has a name and an lcm_bus (and we use it to configure cameras in these notes... see manipulation/models/clutter.scenarios.yaml for an example). https://drake.mit.edu/doxygen_cxx/structdrake_1_1systems_1_1sensors_1_1_camera_config.html
> 
> Moreover, I don't think anything that you add to the diagram is actually specific to realsense? So perhaps we already have enough info to just add the lcm receiver, etc, based on the CameraConfig?
> 
> I do think there could be a tension between the the `sensor_id` and the camera name (which we tend to make more readable). But in that case, maybe we could just bolt on the extra info we need there? e.g. via something like
> 
> ```
> @dc.dataclass
> class CameraConfigWithId(CameraConfig):
>     """A camera configuration that includes the sensor id (for communicating with e.g. the output of a RealSense driver).
> 
>     Args:
>         sensor_id: The camera id for lcm.
>     """
>     sensor_id: str = ""
> ```
> 
> > Reviewed 1 of 1 files at r1, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/RussTedrake/manipulation/315#-:-Nyw4bf_4kQlJZ4mkUQT:b-xs40ft)_ status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @evelyn-fu)

Would it make sense then to use the lcm channel name instead of the sensor_id since that's the only part that's specific to the realsense driver? (The realsense driver names the channels in this specific format: `DRAKE_RGBD_CAMERA_IMAGES_{sensor_id}`)

File: manipulation/station.py [hunk 1]
```diff
@@ -1257,6 +1262,61 @@ def _ApplyDriverConfigsInterface(
...
+    )
+
+    builder.Connect(
+        camera_data_subscriber.get_output_port(),
+        camera_data_receiver.image_array_t_input_port(),
+    )
+
+
+def _ApplyCameraLcmIdsInterface(
+    camera_configs,   # See Scenario.cameras for typing
```
Comment (RussTedrake) [inline manipulation/station.py:1306]: why not make it explicit?  or perhaps even remove this method, and just put the for loop in the _MakeHardwareStationInterface method below?

File: manipulation/station.py [hunk 2]
```diff
@@ -69,6 +72,7 @@
     torque_enabled,
 )
 from pydrake.common.yaml import yaml_load_typed
+from pydrake.common.value import Value
```
Comment (RussTedrake) [inline manipulation/station.py:75]: unused?

File: manipulation/station.py [hunk 3]
```diff
@@ -12,8 +12,10 @@
     lcmt_iiwa_status,
     lcmt_schunk_wsg_command,
     lcmt_schunk_wsg_status,
+    lcmt_image_array,
 )
 from pydrake.all import (
+    AbstractValue,
```
Comment (RussTedrake) [inline manipulation/station.py:18]: unused?

**Review by RussTedrake — CHANGES_REQUESTED** [review]
looks great.  (thanks for the test).  a few very minor requests.

**Review by RussTedrake — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RussTedrake/manipulation/315)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @evelyn-fu)

**Pull request merged**